### PR TITLE
Service accounts: Fix err message suppressed

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What is this feature?**
This makes sure that the error returned from loginConflict is correct and not only ErrUserAlreadyExists

**Special notes for your reviewer**:
not a fix but a bug fix from looking @ a user issue we saw

### CODE REVIEW

File: pkg/services/user/userimpl/user.go [hunk 1]
```diff
@@ -486,7 +486,7 @@ func (s *Service) CreateServiceAccount(ctx context.Context, cmd *user.CreateUser
 
 	err = s.store.LoginConflict(ctx, cmd.Login, cmd.Email, s.cfg.CaseInsensitiveLogin)
```
Comment (eleijonmarck) [inline pkg/services/user/userimpl/user.go:487]: this loginConflict can return both, ErrUserAlreadyExists or ConflictErr

**Review by sakjur — APPROVED** [review]

File: pkg/services/user/userimpl/user.go (hunk 1 above)
Comment (sakjur) [reply pkg/services/user/userimpl/user.go:487]: Why are those different?

Comment (eleijonmarck) [reply pkg/services/user/userimpl/user.go:487]: One error comes from the check for users existing. One is if the users email/login conflicts by casinsensitive. So yeas it's the same error essentially, but we do not know which error is being reported from. Does this make sense? 

Maybe wrapped error would be better here?

Comment (eleijonmarck) [reply pkg/services/user/userimpl/user.go:487]: trying to investigate a issue that a user has from the 4986 support ticket

Comment (eleijonmarck) [reply pkg/services/user/userimpl/user.go:487]: basically we are missing this error - https://github.com/grafana/grafana/blob/7a8f1ab566f6a56a2917e44a3da892c16504054c/pkg/services/user/userimpl/store.go#L307

Comment (sakjur) [reply pkg/services/user/userimpl/user.go:487]: Ah, I understand 🙂

**Pull request merged**